src/wesnoth.cpp

```diff
@@ -872,6 +876,15 @@ static std::vector<std::string> parse_commandline_arguments(std::string input)
 }
 #endif
 
+#ifndef _WIN32
+void wesnoth_terminate_handler(int sig);
+void wesnoth_terminate_handler(int sig) {
+	SDL_Event event;
+	event.type = TERMINATE_EVENT;
```

Comment:
This could equally easy be SDL_QUIT instead of the user event, making the user event unnecessary.